validation/genesis/genesis-allocs_test.go

```diff
@@ -35,12 +34,15 @@ func TestMain(m *testing.M) {
 	thisDir := filepath.Dir(filename)
 	temporaryOptimismDir = path.Join(thisDir, "../../../optimism-temporary")
 
-	// Clone the repo if the folder doesn't exist
+	// Clone the repo if the folder doesn't exist, otherwise pull the latest changes
 	_, err := os.Stat(temporaryOptimismDir)
 	needToClone := os.IsNotExist(err)
 	if needToClone {
 		mustExecuteCommandInDir(thisDir,
 			exec.Command("git", "clone", "--recurse-submodules", "https://github.com/ethereum-optimism/optimism.git", temporaryOptimismDir))
+	} else {
+		mustExecuteCommandInDir(thisDir,
+			exec.Command("git", "pull", temporaryOptimismDir))
```

Comment:
Do we want a full clone or would a (much faster) shallow suffice?

Reply:
I also had that thought, but it was already doing a full clone seemingly bc it needs to check out arbitrary commits across multiple tests; potentially we could shallow clone at the specified commit each time but I'm not fully sure if that would break anything